client/cpp/synnax/channel/channel.cpp

```diff
@@ -0,0 +1,123 @@
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "synnax/channel/channel.h"
+#include <grpcpp/grpcpp.h>
+#include "v1/channel.pb.h"
+#include "synnax/exceptions.h"
+#include "synnax/telem/telem.h"
+
+using namespace Synnax;
+
+std::string RETRIEVE_ENDPOINT = "/channel/retrieve";
+std::string CREATE_ENDPOINT = "/channel/create";
+
+
+Channel::Channel(const api::v1::Channel &ch) {
+    name = ch.name();
+    key = ch.key();
+    index = ch.index();
+    rate = Telem::Rate(ch.rate());
+    is_index = ch.is_index();
+    leaseholder = ch.leaseholder();
+    dataType = Telem::DataType(ch.data_type());
+}
+
+Channel::Channel(
+        std::string name,
+        Telem::DataType dataType,
+        Telem::Rate rate,
+        bool is_index,
+        std::uint32_t leaseholder,
+        ChannelKey index,
+        ChannelKey key
+) : dataType(dataType),
+    name(name),
+    key(key),
+    index(index),
+    rate(rate),
+    is_index(is_index),
+    leaseholder(leaseholder) {}
+
+
+void Channel::to_proto(api::v1::Channel *a) const {
+    a->set_name(name);
+    a->set_data_type(dataType.value);
+    a->set_rate(rate.value);
+    a->set_is_index(is_index);
+    a->set_leaseholder(leaseholder);
+    a->set_index(index);
+    a->set_key(key);
+}
+
+
+Channel ChannelClient::create(
+        std::string name,
+        Telem::DataType data_type,
+        Telem::Rate rate,
+        ChannelKey index,
+        bool is_index
+) {
+    auto req = api::v1::ChannelCreateRequest();
+    auto a = req.add_channels();
+    a->set_name(name);
+    a->set_index(index);
+    a->set_is_index(is_index);
+    a->set_data_type(data_type.value);
+    a->set_rate(rate.value);
+    auto response = create_client->send(CREATE_ENDPOINT, req);
+    return {response.first.channels(0)};
+}
+
+void ChannelClient::create(std::vector<Channel> &channels) {
+    auto req = api::v1::ChannelCreateRequest();
+    for (const auto &ch: channels) {
+        auto a = req.add_channels();
```

Comment:
The type of a is hard to infer here, and usually you only use auto when the type is easy to deduce, like

`auto a = make_unique<int>(42);`

Had to look up that this was a pointer and whatnot. Otherwise this is clean code, like the flow of passing the element pointer to `to_proto` and setting the required values, seems efficient.

Reply:
Fixed.